awc/src/response.rs

```diff
@@ -21,6 +24,43 @@ use crate::error::JsonPayloadError;
 pub struct ClientResponse<S = PayloadStream> {
     pub(crate) head: ResponseHead,
     pub(crate) payload: Payload<S>,
+    pub(crate) timeout: ResponseTimeout,
+}
+
+// a helper enum for response timeout for reusing the boxed sleep.
+// It's pass from `SendClientRequest`.
```

Comment:
despite it being private, please make all possible comments into doc comments so they show up on symbol hover hints

Reply:
removed. It's just a hint to people really want to looking at the code why a boxed timer is passed around. I guess it's easy enough to understand so I removed them.

Reply:
i would say (having not looked at the code yet) it's not particularly ovious why the Disbled variant contains a timer too